src/crypto/CHIPCryptoPAL.h

```diff
@@ -228,68 +228,120 @@ class ECPKey
+     */
+    size_t Length() const { return mLength; }
 
-    /** @brief Returns max capacity of the buffer
-     **/
-    static constexpr size_t Capacity() { return sizeof(bytes); }
+    /**
+     * @brief Returns non-const pointer to start of the underlying buffer
+     */
+    uint8_t * Bytes() { return &mBytes[0]; }
 
-    /** @brief Returns pointer to start of underlying buffer
-     **/
-    uint8_t * Bytes() { return &bytes[0]; }
+    /**
+     * @brief Returns const pointer to start of the underlying buffer
+     */
+    const uint8_t * ConstBytes() const { return &mBytes[0]; }
 
-    /** @brief Returns const pointer to start of underlying buffer
-     **/
-    const uint8_t * ConstBytes() const { return &bytes[0]; }
+    /**
+     * @brief Constructs fixed span from the underlying buffer
+     */
+    ByteSpan Span() const { return ByteSpan(ConstBytes(), Length()); }
 
-    /** @brief Returns buffer pointer
-     **/
-    operator uint8_t *() { return bytes; }
-    operator const uint8_t *() const { return bytes; }
+    /**
+     * @brief Returns capacity of the buffer
+     */
+    static constexpr size_t Capacity() { return kCapacity; }
 
 private:
-    uint8_t bytes[Cap];
-    size_t length = 0;
+    uint8_t mBytes[kCapacity];
+    size_t mLength = 0;
 };
 
-typedef CapacityBoundBuffer<kMax_ECDSA_Signature_Length> P256ECDSASignature;
+template <size_t kCapacity>
+class SensitiveDataFixedBuffer
+{
+public:
+    SensitiveDataFixedBuffer() = default;
+
+    constexpr explicit SensitiveDataFixedBuffer(const uint8_t (&rawValue)[kCapacity])
+    {
+        memcpy(&mBytes[0], &rawValue[0], kCapacity);
+    }
+
+    constexpr explicit SensitiveDataFixedBuffer(const FixedByteSpan<kCapacity> & value)
+    {
+        memcpy(&mBytes[0], value.data(), kCapacity);
+    }
+
+    ~SensitiveDataFixedBuffer()
+    {
+        // Sanitize after use
+        ClearSecretData(mBytes);
+    }
 
-typedef CapacityBoundBuffer<kMax_ECDH_Secret_Length> P256ECDHDerivedSecret;
+    /**
+     * @brief Returns fixed length of the buffer
+     */
+    constexpr size_t Length() const { return kCapacity; }
+
+    /**
+     * @brief Returns non-const pointer to start of the underlying buffer
+     */
+    uint8_t * Bytes() { return &mBytes[0]; }
+
+    /**
+     * @brief Returns const pointer to start of the underlying buffer
+     */
+    const uint8_t * ConstBytes() const { return &mBytes[0]; }
+
+    /**
+     * @brief Constructs span from the underlying buffer
+     */
+    FixedByteSpan<kCapacity> Span() const { return FixedByteSpan<kCapacity>(mBytes); }
+
+private:
+    uint8_t mBytes[kCapacity];
+};
+
+using P256ECDSASignature    = SensitiveDataBuffer<kMax_ECDSA_Signature_Length>;
+using P256ECDHDerivedSecret = SensitiveDataBuffer<kMax_ECDH_Secret_Length>;
+
+using IdentityProtectionKey     = SensitiveDataFixedBuffer<CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES>;
+using IdentityProtectionKeySpan = FixedByteSpan<Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES>;
```

Comment:
Ideally, we would get rid of `IdentityProtectionKeySpan` entirely and pass around references to `IdentityProtectionKey`...  Followup issue for that, please.

Reply:
https://github.com/project-chip/connectedhomeip/issues/24428